Tests/NIOHTTP1Tests/HTTPHeadersTest.swift

```diff
@@ -179,4 +179,74 @@ class HTTPHeadersTest : XCTestCase {
+                        value: HTTPHeaderIndex(start: 12, length: 17)),
+             HTTPHeader(name: HTTPHeaderIndex(start: 32, length: 12),
+                        value: HTTPHeaderIndex(start: 46, length: 9)),
+             HTTPHeader(name: HTTPHeaderIndex(start: 57, length: 10),
+                        value: HTTPHeaderIndex(start: 69, length: 17))],
+                                  keepAliveState: .unknown)
+        
+        XCTAssertTrue(headers.isKeepAlive(version: HTTPVersion(major: 1, minor: 1)))
+    }
+
+    func testKeepAliveStateHasClose() {
+        var buffer = ByteBufferAllocator().buffer(capacity: 32)
+        buffer.write(string: "Connection: x-options,  other\r\n" +
+                             "Content-Type: text/html\r\n" +
+                             "Connection: server,     clOse")
+        let headers = HTTPHeaders(buffer: buffer, headers:
+            [HTTPHeader(name: HTTPHeaderIndex(start: 0, length: 10),
+                        value: HTTPHeaderIndex(start: 12, length: 17)),
+             HTTPHeader(name: HTTPHeaderIndex(start: 32, length: 12),
+                        value: HTTPHeaderIndex(start: 46, length: 9)),
+             HTTPHeader(name: HTTPHeaderIndex(start: 56, length: 10),
+                        value: HTTPHeaderIndex(start: 68, length: 17))],
+                                  keepAliveState: .unknown)
+        
+        XCTAssertFalse(headers.isKeepAlive(version: HTTPVersion(major: 1, minor: 1)))
+    }
+    
+    func testResolveNonContiguousHeaders() {
+        var buffer = ByteBufferAllocator().buffer(capacity: 32)
+        buffer.write(string: "Connection: x-options,  other\r\n" +
+                             "Content-Type: text/html\r\n" +
+                             "Connection: server,     close")
+        let headers =
+            [HTTPHeader(name: HTTPHeaderIndex(start: 0, length: 10),
+                        value: HTTPHeaderIndex(start: 12, length: 17)),
+             HTTPHeader(name: HTTPHeaderIndex(start: 32, length: 12),
+                        value: HTTPHeaderIndex(start: 46, length: 9)),
+             HTTPHeader(name: HTTPHeaderIndex(start: 56, length: 10),
+                        value: HTTPHeaderIndex(start: 68, length: 17))]
+        
+        var tokenSource = HTTPListHeaderIterator(
+            headerName: "Connection".utf8, headers: HTTPHeaders.init(buffer: buffer, headers: headers, keepAliveState: .unknown))
+        
+        var currentToken = tokenSource.next()
+        XCTAssertEqual(readAllTheStringFromCollection(currentToken), "x-options")
+        currentToken = tokenSource.next()
+        XCTAssertEqual(readAllTheStringFromCollection(currentToken), "other")
+        currentToken = tokenSource.next()
+        XCTAssertEqual(readAllTheStringFromCollection(currentToken), "server")
+        currentToken = tokenSource.next()
+        XCTAssertEqual(readAllTheStringFromCollection(currentToken), "close")
```

Comment:
You can replace `readAllTheStringFromCollection` with just calling `String(decoding: currentToken, as: UTF8.self)`.